bldsys/cmake/component_helpers.cmake

```diff
@@ -16,21 +16,37 @@
 
 add_custom_target(vkb_components)
 
+set_property(TARGET vkb_components PROPERTY FOLDER "components")
+
+function(vkb__register_headers)
+    set(options)
+    set(oneValueArgs FOLDER OUTPUT)
+    set(multiValueArgs HEADERS)
+
+    cmake_parse_arguments(TARGET "${options}" "${oneValueArgs}" "${multiValueArgs}" ${ARGN})
+
+    set(TEMP ${TARGET_HEADERS})
+    list(TRANSFORM TEMP PREPEND "${CMAKE_CURRENT_SOURCE_DIR}/include/components/${TARGET_FOLDER}/")
+    set(${TARGET_OUTPUT} ${TEMP} PARENT_SCOPE)
+endfunction()
+
 function(vkb__register_component)
     set(options)
     set(oneValueArgs NAME)
-    set(multiValueArgs SRC LINK_LIBS INCLUDE_DIRS)
+    set(multiValueArgs SRC HEADERS LINK_LIBS INCLUDE_DIRS)
 
     cmake_parse_arguments(TARGET "${options}" "${oneValueArgs}" "${multiValueArgs}" ${ARGN})
 
     if(TARGET_NAME STREQUAL "")
         message(FATAL_ERROR "NAME must be defined in vkb__register_tests")
     endif()
 
-    if(TARGET_SRC) ## Create static library
+    if(TARGET_SRC) # Create static library
         message("ADDING STATIC: vkb__${TARGET_NAME}")
 
-        add_library("vkb__${TARGET_NAME}" STATIC ${TARGET_SRC})
+        add_library("vkb__${TARGET_NAME}" STATIC ${TARGET_SRC} ${TARGET_HEADERS})
```

Comment:
think this is the same as in 490?  Not sure of the advantage of adding headers explicitly to the cmake target.